src/snowflake/cli/api/project/schemas/identifier_model.py

```diff
@@ -0,0 +1,29 @@
+from __future__ import annotations
+
+from typing import Optional, cast
+
+from pydantic import Field
+from snowflake.cli.api.project.schemas.updatable_model import IdentifierField
+
+
+class ObjectIdentifierBaseModel:
```

Comment:
nit: please add a doc string to state what this represents. It would help to understand the little bit of metaprogramming below IMHO.